modules/platforms/dotnet/release/Program.cs

```diff
@@ -0,0 +1,99 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Apache.Ignite.Core;
+using Apache.Ignite.Core.Cache.Configuration;
+using Apache.Ignite.Core.Client;
+using Apache.Ignite.Core.Discovery.Tcp;
+using Apache.Ignite.Core.Discovery.Tcp.Static;
+using Apache.Ignite.Linq;
+
+namespace test_proj
+{
+    public static class Program
+    {
+        public static void Main()
+        {
+            // Don't use async Main - compatibility with C# 7.0
+            MainAsync().GetAwaiter().GetResult();
+        }
+
+        private static async Task MainAsync()
+        {
+            var cfg = new IgniteConfiguration
+            {
+                DiscoverySpi = new TcpDiscoverySpi
+                {
+                    IpFinder = new TcpDiscoveryStaticIpFinder
+                    {
+                        Endpoints = new[] {"127.0.0.1:47500"}
+                    },
+                    SocketTimeout = TimeSpan.FromSeconds(0.3)
+                }
+            };
+
+            using (var ignite = Ignition.Start(cfg))
+            {
+                var cacheCfg = new CacheConfiguration(
+                    "cache1",
+                    new QueryEntity(typeof(int), typeof(Person)));
+                
+                var cache = ignite.CreateCache<int, Person>(cacheCfg);
+                
+                cache.Put(1, new Person(1));
+                Debug.Assert(1 == cache[1].Age);
+
+                var resPerson = cache.AsCacheQueryable()
+                    .Where(e => e.Key > 0 && e.Value.Name.StartsWith("Person"))
+                    .Select(e => e.Value)
+                    .Single();
+                Debug.Assert(1 == resPerson.Age);
+
+                using (var igniteThin = Ignition.StartClient(new IgniteClientConfiguration("127.0.0.1")))
+                {
+                    var cacheThin = igniteThin.GetCache<int, Person>(cacheCfg.Name);
+                    var personThin = await cacheThin.GetAsync(1);
+                    Debug.Assert("Person-1" == personThin.Name);
+
+                    var personNames = cacheThin.AsCacheQueryable()
+                        .Where(e => e.Key != 2 && e.Value.Age < 10)
+                        .Select(e => e.Value.Name)
+                        .ToArray();
+                    Debug.Assert(personNames.SequenceEqual(new[] {"Person-1"}));
+                }
+            }
+        }
+    }
+
+    public class Person
+    {
+        public Person(int age)
+        {
+            Age = age;
+            Name = $"Person-{age}";
```

Comment:
Wow, didn't know .NET can do that.

Reply:
Killer feature, right? 😁 
There is lots of impressive stuff in recent versions, but we are stuck with 4.0 in the main code base for now :(
This script uses modern tooling (.NET Core), so we can use newer language features.